lib/server.js

```diff
@@ -24,6 +26,16 @@ const initServer = function(
     applications = apps.createAppsIndex(applications);
   }
 
+  // versions cached for efficient search when provided version is a range
+  this._cachedVersions = new Map();
+  applications.forEach((appVersions, appName) => {
+    const versions = Array.from(appVersions)
+      .filter(app => app[0] !== 'latest')
+      .map(version => [new semver.SemVer(version[0]), version[1]])
```

Comment:
Yes, but here it is obvious, because of line 32: this is an array created from the map with versions as keys, and applications as values, there is no need to make a destruction assignment here.
However, I do agree that arguments in lambdas should be renamed.